src/cc/frontends/p4/compiler/target.py

```diff
@@ -0,0 +1,204 @@
+        self.targetName = target
+
+    def getIncludes(self):
+        return ""
+
+    def serializeLookup(self, serializer, tableName, key, value):
+        serializer.appendFormat("{0} = bpf_map_lookup_elem(&{1}, &{2});",
+                                value, tableName, key)
+
+    def serializeUpdate(self, serializer, tableName, key, value):
+        serializer.appendFormat("bpf_map_update_elem(&{0}, &{1}, &{2}, BPF_ANY);",
+                                tableName, key, value)
+
+    def serializeLicense(self, serializer, licenseString):
+        assert isinstance(serializer, ProgramSerializer)
+        serializer.emitIndent()
+        serializer.appendFormat("char _license[] {0}(\"license\") = \"{1}\";", self.config.section, licenseString)
+        serializer.newline()
+
+    def serializeCodeSection(self, serializer):
+        assert isinstance(serializer, ProgramSerializer)
+        serializer.appendFormat("{0}(\"{1}\")", self.section, self.entrySection)
+
+    def serializeTableDeclaration(self, serializer, tableName, isHash, keyType, valueType, size):
+        assert isinstance(serializer, ProgramSerializer)
+        assert isinstance(tableName, str)
+        assert isinstance(isHash, bool)
+        assert isinstance(keyType, str)
+        assert isinstance(valueType, str)
+        assert isinstance(size, int)
+
+        serializer.emitIndent()
+        serializer.appendFormat("struct {0} {1}(\"maps\") {2} = ",
+                                self.tableName, self.section, tableName)
+        serializer.blockStart()
+
+        serializer.emitIndent()
+        serializer.append(".type = ")
+        if isHash:
+            serializer.appendLine("BPF_MAP_TYPE_HASH,")
+        else:
+            serializer.appendLine("BPF_MAP_TYPE_ARRAY,")
+
+        serializer.emitIndent()
+        serializer.appendFormat(".{0} = sizeof(struct {1}), ", self.tableKeyAttribute, keyType)
+        serializer.newline()
+
+        serializer.emitIndent()
+        serializer.appendFormat(".{0} = sizeof(struct {1}), ", self.tableValueAttribute, valueType)
+        serializer.newline()
+
+        serializer.emitIndent()
+        serializer.appendFormat(".{0} = {1}, ", self.tableSizeAttribute, size)
+        serializer.newline()
+
+        serializer.blockEnd(False)
+        serializer.endOfStatement(True)
+
+    def generateDword(self, serializer):
+        serializer.appendFormat("static inline {0}64 load_dword(void *skb, {0}64 off)", self.uprefix)
+        serializer.newline()
+        serializer.blockStart()
+        serializer.emitIndent()
+        serializer.appendFormat("return (({0}64)load_word(skb, off) << 32) | load_word(skb, off + 4);",
+                                self.uprefix)
+        serializer.newline()
+        serializer.blockEnd(True)
+
+
+
+# Represents a target that attaches on a socket
+class SocketConfig(TargetConfig):
+    def __init__(self):
+        TargetConfig.__init__(self, "Socket")
+        self.entrySection = "socket1"
```

Comment:
Can you explain what this does? It's not clear to me, and my gut reaction is this is something that only makes sense in a non-BCC workflow, where elf header sections are used.